File: Leetcode/1413.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int minVal = numeric_limits<int>::max();
        int initial = 0;
        for (const auto &i : nums) {
            initial += i;
            minVal = min(minVal, initial);
        }
        if (minVal >= 0) {
            return 1;
        }
        return abs(minVal) + 1;
    }
};

File: Leetcode/1413/1413.go
func minStartValue(nums []int) int {
	prefixSum := nums[0]
	minPrefixSum := prefixSum
	for i := 1; i < len(nums); i++ {
		prefixSum += nums[i]
		if prefixSum < minPrefixSum {
			minPrefixSum = prefixSum
		}
	}
	if minPrefixSum > 0 {
		return 1
	}
	return -minPrefixSum + 1
}